### ISSUE TITLE
Any plan for docker?

### ISSUE THREAD

**Issue by yangdongsheng** [huawei-openlab/oct#354]
Hi guys, I found docker is not supported currently, do you have any plan for it?

Thanx

**Comment by yangdongsheng** [comment]
Is that difficult now?

**Comment by zenlinTechnofreak** [comment]
@yangdongsheng  Yes, It is in the plan, and do u have any advice for docker supporting.

**Comment by yangdongsheng** [comment]
@zenlinTechnofreak Great!! Is there any branch I can refer to? :)

**Comment by yangdongsheng** [comment]
@zenlinTechnofreak BTW, I found the links for "Testing Flow" are all outdated. Where can I get these docs now? thanx a lot

**Comment by zenlinTechnofreak** [comment]
@yangdongsheng We are talking about the detail realization way with OCI , and not start coding yet, because the version of OCI is not very stable yet.
Anyway, will anounce to you once we have a branch for it.
By the way , why you concern about this?

Welcome any suggestion and advice, u also can give it to the [Google Groups](https://groups.google.com/a/opencontainers.org/forum/#!forum/dev), we have a topic there,
[OCT](https://groups.google.com/a/opencontainers.org/forum/#!topic/dev/T7c8ofdQpo0)

**Comment by zenlinTechnofreak** [comment]
@yangdongsheng Yes, the link is outdated, you can just use the README.md for using the test temporally
And we are drawing a new picture for it.

**Comment by zenlinTechnofreak** [comment]
@yangdongsheng the main flow is,
- Read config from cases.conf
- Using ocitool to generate runtime.json and config.json
- Using ocitool to generate containerend testing programme, called runtimetest
- Compose the rootfs, 2 json files, runtimetest to a bundle, each bundle corresponds to one testunit.
- Convert bundle to each image format for testing runtime if necessary.
- Run runtime for testing.
- Output result.

**Comment by yangdongsheng** [comment]
@zenlinTechnofreak Haha, test flow sounds great. But, docker seems using a different config format with spec said, E.g, there is no Rootfs in config.json after docker create, then we would get a fail?

**Comment by zenlinTechnofreak** [comment]
@yangdongsheng Yes, different runtime have different image fomat, what we lack of is a tool like oci2docker, which can supporting convert oci bundle to docker image.
The way like supporting RKT.

And if runc can support docker to run oci bundle directly, it should be very nice.

**Comment by yangdongsheng** [comment]
@zenlinTechnofreak oci2docker? Ha, it addressed my question in my mind now. Thanx a lot for the information. We are curious about making docker to follow OCI/spec. And glad to use OCT for our testing. that's helpful. :)

**Comment by zenlinTechnofreak** [comment]
@yangdongsheng 
Nice, 
to ”We are curious about making docker to follow OCI/spec“, do u have any considering about it?

**Comment by yangdongsheng** [comment]
@zenlinTechnofreak We are doing investigation now. At first, we want to know "What we need to do for docker to make OCI/spec happy?". Then we want to run a OCT for docker. Therefore, I come here to open a issue about "Any plan for docker" :).

**Comment by yangdongsheng** [comment]
@zenlinTechnofreak BTW, is there any plan to contribute OCT to be a project in OCI? Is there a schedule? And what's the opinion from OCI guys? As we are curious about OCT, we want to know what's the final goal of OCT. Thanx in advance. :)

**Comment by zenlinTechnofreak** [comment]
@yangdongsheng 
The final goal of OCT is pretty same with OCI, we want to make containers more open and normalize, supporting OCI to create open industry standards.
In my opinion, the future of container should consist of one open industry standard, different container runtimes and  a testing framework for validating the compliant of runtime to the industry standard. What OCT want to do is creating the testing framework and testing package in OCI.

OCT is in the roadmap of specs 1.0 now.

Are you interesting in developing OCT? To make the future of container more beautiful, together with us.
We can disscuss the design of OCT or oci2docker in  dev@opencontainers.org, in topic oct.

**Comment by zenlinTechnofreak** [comment]
@yangdongsheng The testflow chart of OCT have been updated, thanks.

**Comment by liangchenye** [comment]
@yangdongsheng @Mashimiao 
Maybe you already know, OCT is now a part of opencontainers/ocitools.
The testing flow is not merged yet. Once that was done, we can push more
codes to that project. Comparing to OCI project, OCT will remain flexible.

**Comment by Mashimiao** [comment]
@liangchenye  got it, thanks. By the way, when will OCT be merged into ocitools?
